src/AutoRest.CSharp/Common/Output/Models/FieldDeclaration.cs

```diff
@@ -9,7 +9,7 @@
 
 namespace AutoRest.CSharp.Output.Models
 {
-    internal record FieldDeclaration(FormattableString? Description, FieldModifiers Modifiers, CSharpType Type, CSharpType ValueType, CodeWriterDeclaration Declaration, FormattableString? DefaultValue, bool IsRequired, SerializationFormat SerializationFormat, bool IsField = false, bool WriteAsProperty = false, bool OptionalViaNullability = false, FieldModifiers? GetterModifiers = null, FieldModifiers? SetterModifiers = null, SourcePropertySerializationMapping? SerializationMapping = null)
+    internal record FieldDeclaration(FormattableString? Description, FieldModifiers Modifiers, CSharpType Type, CSharpType ValueType, CodeWriterDeclaration Declaration, FormattableString? InitializationValue, bool IsRequired, SerializationFormat SerializationFormat, bool IsField = false, bool WriteAsProperty = false, bool OptionalViaNullability = false, FieldModifiers? GetterModifiers = null, FieldModifiers? SetterModifiers = null, SourcePropertySerializationMapping? SerializationMapping = null)
```

Comment:
default value is a little bit confusing because here is not a spec of something, it is not really a default value - more like an initialization value for a field which the class use to initialize the corresponding instance.
I changed the name here to align with the name in `ObjectTypeProperty`.

Reply:
The property `writeAsProperty` is confusing as well.
If the member is a field we should construct an instance of FieldDeclaration, and if the member is a property, we should construct an instance of corresponding to property.
We should really do not have a thing which is "field declaration", but writes as a property.